core/src/consensus/stake/actions.rs

```diff
@@ -50,10 +56,19 @@ pub enum Action {
         params: Box<CommonParams>,
         signatures: Vec<Signature>,
     },
+    ReportDoubleVote {
+        message1: M,
+        message2: M,
+    },
 }
 
-impl Action {
-    pub fn verify(&self, current_params: &CommonParams) -> Result<(), SyntaxError> {
+impl<M: Message> Action<M> {
+    pub fn verify(
+        &self,
+        current_params: &CommonParams,
+        client: Option<Arc<ConsensusClient>>,
```

Comment:
You could've accept just `&ConsensusClient`. Anyway, We'll ship it now and fix it later.